portfolio/src/main/webapp/script.js

```diff
@@ -13,65 +13,72 @@
-  fetch(url).then(response => response.json()).then((messages) => {
-    const messagesListElement = document.getElementById('messages-container');
+  return fetch(url).then(response =>response.json());
+}
+
+async function handleGetCommentsClick(){
+   renderLoadingListElement();
+   const messages = await getCommentsFromServer();
+   renderComments(messages);
+}
+
+function renderComments(messages){
+    const messagesListElement = document.getElementById('comments-container');
     if(messagesListElement.innerHTML !== ''){
         messagesListElement.innerHTML = '';
     }
 
-    for(i = 0; i < messages.length; ++i){
-        childToAppend = createListElement(messages[i].propertyMap.email, messages[i].propertyMap.content);
-        childToAppend.appendChild(createButton(messages[i].propertyMap.messageId));
-        messagesListElement.appendChild(childToAppend);
+    for(var i = 0; i < messages.length; ++i){
+        var domListElement = createDOMListElement(messages[i].propertyMap.email, messages[i].propertyMap.content);
+        domListElement.appendChild(createDOMButton(messages[i].propertyMap.messageId));
+        messagesListElement.appendChild(domListElement);
     }
     
     if(messages.length === 0){
-        var content = "No messages available!";
-        childToAppend = createListElement(content);
-        messagesListElement.appendChild(childToAppend);
+        var content = "No comments available!";
+        var domListElement = createDOMListElement(content);
+        messagesListElement.appendChild(domListElement);
     }
-  });
 }
 
-function createListElement(email, text) {
-  const liElement = document.createElement('li');
-  if(email != "")
-   liElement.innerText = email + ": ";
-  liElement.innerText += text;
-  liElement.classList.add('list-group-item');
-  return liElement;
+function createDOMListElement(email, text) {
+  const domListElement = document.createElement('li');
+  if(email !== "")
+   domListElement.innerText = email + ": ";
+  domListElement.innerText += text;
+  domListElement.classList.add('list-group-item');
+  return domListElement;
 }
 
-function createButton(messageId) {
-  const buttonElement = document.createElement('button');
-  buttonElement.classList.add('delete-button');
-  buttonElement.textContent = "Delete";
-  buttonElement.id = messageId;
-  buttonElement.setAttribute( "onClick", "javascript: deleteMessage(this);" );
-  return buttonElement;
+function createDOMButton(messageId) {
+  const domButtonElement = document.createElement('button');
+  domButtonElement.classList.add('delete-button');
+  domButtonElement.textContent = "Delete";
+  domButtonElement.id = messageId;
+  domButtonElement.setAttribute( "onClick", "javascript: handleDeleteCommentClick(this);" );
+  return domButtonElement;
 }
 
+function handleDeleteCommentClick(buttonElement){
+   deleteComment(buttonElement);
+   const messages = getCommentsFromServer();
+   renderComments(messages);
```

Comment:
It looks like `getCommentsFromServer` returns a `Promise`. You can use the `then` method to chain your calls, or use the async/await combination that you're also using at `handleGetCommentsClick`, both are good (async/await is the modern approach)

```
getCommentsFromServer().then(renderComments);
```